tests/test-team/template-mgmt-component-tests/template-mgmt-manage-templates-page.component.ts

```diff
@@ -0,0 +1,298 @@
+    message: 'test example message',
+    subject: 'test example subject',
+    templateType: TemplateType.EMAIL,
+    templateStatus: TemplateStatus.NOT_YET_SUBMITTED,
+    createdAt: '2010-10-11T10:10:10.100Z',
+  }),
+  smsSubmitted: TemplateFactory.create({
+    id: uuid(),
+    name: 'sms-submitted_manage-templates-page',
+    message: 'test example message',
+    templateType: TemplateType.SMS,
+    templateStatus: TemplateStatus.SUBMITTED,
+    createdAt: '2010-10-10T11:11:11.111Z',
+  }),
+  smsNotYetSubmitted: TemplateFactory.create({
+    id: uuid(),
+    name: 'sms-not-yet-submitted_manage-templates-page',
+    message: 'test example message',
+    templateType: TemplateType.SMS,
+    templateStatus: TemplateStatus.NOT_YET_SUBMITTED,
+    createdAt: '2010-10-10T10:10:10.100Z',
+  }),
+  nhsAppSubmitted: TemplateFactory.create({
+    id: uuid(),
+    name: 'nhs-app-submitted_manage-templates-page',
+    message: 'test example message',
+    templateType: TemplateType.NHS_APP,
+    templateStatus: TemplateStatus.SUBMITTED,
+    createdAt: '2010-10-09T11:11:11.111Z',
+  }),
+  nhsAppNotYetSubmitted: TemplateFactory.create({
+    id: uuid(),
+    name: 'nhs-app-not-yet-submitted_manage-templates-page',
+    message: 'test example message',
+    templateType: TemplateType.NHS_APP,
+    templateStatus: TemplateStatus.NOT_YET_SUBMITTED,
+    createdAt: '2010-10-09T10:10:10.100Z',
+  }),
+};
+
+test.describe('Manage templates page', () => {
+  const templateStorageHelper = new TemplateStorageHelper(
+    Object.values(templates)
+  );
+
+  test.beforeAll(async () => {
+    await templateStorageHelper.seedTemplateData();
+  });
+
+  test.afterAll(async () => {
+    await templateStorageHelper.deleteTemplateData();
+  });
+
+  test('common page tests', async ({ page, baseURL }) => {
+    const props = {
+      page: new ManageTemplatesPage(page),
+      baseURL,
+    };
+
+    await assertSkipToMainContent(props);
+    await assertNotifyBannerLink(props);
+    await assertLoginLink(props);
+    await assertFooterLinks(props);
+    await assertGoBackLinkNotPresent(props);
+  });
+
+  test('should navigate to the manage templates page', async ({
+    page,
+    baseURL,
+  }) => {
+    const manageTemplatesPage = new ManageTemplatesPage(page);
+    await manageTemplatesPage.loadPage();
+    await expect(page).toHaveURL(`${baseURL}/templates/manage-templates`);
+    expect(await manageTemplatesPage.pageHeader.textContent()).toBe(
+      'Message templates'
+    );
+    await expect(manageTemplatesPage.createTemplateButton).toBeVisible();
+  });
+
+  test('Submitted template items have correct status indicator', async ({
+    page,
+    baseURL,
+  }) => {
+    const manageTemplatesPage = new ManageTemplatesPage(page);
+    await manageTemplatesPage.loadPage();
+
+    expect(page.url()).toContain(`${baseURL}/templates/manage-templates`);
+
+    const email = page.locator(
+      'tr:has-text("email-submitted_manage-templates-page")'
```

Comment:
Could these be properties on  the `ManageTemplatesPage` class?

Reply:
I've just tried it, but then decided they are probably best where they are. These locators are specific to this test file and the templates loaded into the manage templates page.